be/src/exprs/vectorized/json_functions.cpp

```diff
@@ -8,29 +8,45 @@
 #include <boost/algorithm/string.hpp>
 #include <boost/tokenizer.hpp>
 
+#include "column/column_builder.h"
 #include "column/column_helper.h"
 #include "column/column_viewer.h"
 #include "common/status.h"
+#include "exprs/vectorized/function_helper.h"
+#include "exprs/vectorized/jsonpath.h"
+#include "glog/logging.h"
+#include "gutil/strings/split.h"
 #include "gutil/strings/substitute.h"
+#include "udf/udf.h"
+#include "util/json.h"
+#include "velocypack/vpack.h"
 
 namespace starrocks::vectorized {
 
 // static const re2::RE2 JSON_PATTERN("^([a-zA-Z0-9_\\-\\:\\s#\\|\\.]*)(?:\\[([0-9]+)\\])?");
 // json path cannot contains: ", [, ]
-static const re2::RE2 JSON_PATTERN(R"(^([^\"\[\]]*)(?:\[([0-9]+|\*)\])?)");
+const re2::RE2 JSON_PATTERN(R"(^([^\"\[\]]*)(?:\[([0-9\:\*]+)\])?)");
 
 Status JsonFunctions::_get_parsed_paths(const std::vector<std::string>& path_exprs,
-                                        std::vector<JsonPath>* parsed_paths) {
-    if (path_exprs[0] != "$") {
-        parsed_paths->emplace_back("", -1, false);
-        return Status::InvalidArgument(strings::Substitute("Invalid json path: $0", path_exprs[0]));
-    } else {
-        parsed_paths->emplace_back("$", -1, true);
-    }
+                                        std::vector<SimpleJsonPath>* parsed_paths) {
+    // Allow two kind of syntax:
+    // strict jsonpath: $.a.b[x]
+    // simple syntax: a.b
 
-    for (int i = 1; i < path_exprs.size(); i++) {
+    for (int i = 0; i < path_exprs.size(); i++) {
         std::string col;
         std::string index;
+        auto& current = path_exprs[i];
+
+        if (i == 0) {
+            if (current != "$") {
+                parsed_paths->emplace_back("", -1, true);
+            } else {
+                parsed_paths->emplace_back("$", -1, true);
+                continue;
+            }
+        }
+
         if (UNLIKELY(!RE2::FullMatch(path_exprs[i], JSON_PATTERN, &col, &index))) {
```

Comment:
`index` may be `"1:3"`, why does the `else` branch always consider it as an integer? Or maybe we should use different `JSON_PATTERN` between here and `JsonPathPiece::parse`?

Reply:
Well, this function only deal with SimpleJsonPath, which does not support `arr[1:3]` syntax.

The `SimpleJsonPath` is previous implementation, and the `JsonPath` is yet another implementation, covers more complex jsonpath syntax like `arr[1:3]`, `arr[1].xxx`, `arr[*].xxx`.

Reply:
@mofeiatwork This function only deals with `SimpleJsonPath`, but also uses `JSON_PATTERN(R"(^([^\"\[\]]*)(?:\[([0-9\:\*]+)\])?)")`, which also matches the string like `$.a.b[1:3]`. Therefore, should we use a different `JSON_PATTERN` here?

Reply:
Good idea. I will fix it soon.